lib/wallet/http.js

```diff
@@ -833,6 +784,441 @@ class HTTP extends Server {
+
+      if (!ns)
+        return res.json(404);
+
+      const bids = await req.wallet.getBidsByName(name);
+      const reveals = await req.wallet.getRevealsByName(name);
+
+      const info = ns.getJSON(height, network);
+      info.bids = [];
+      info.reveals = [];
+
+      for (const bid of bids)
+        info.bids.push(bid.toJSON());
+
+      for (const reveal of reveals)
+        info.reveals.push(reveal.toJSON());
+
+      return res.json(200, info);
+    });
+
+    // All Wallet Bids
+    this.get('/wallet/:id/bid', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const own = valid.bool('own', false);
+
+      const bids = await req.wallet.getBidsByName();
+      const items = [];
+
+      for (const bid of bids) {
+        if (!own || bid.own)
+          items.push(bid.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Wallet Bids by Name
+    this.get('/wallet/:id/bid/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      let own = valid.bool('own', false);
+
+      if (name)
+        assert(rules.verifyName(name), 'Must pass valid name.');
+
+      if (!name)
+        own = true;
+
+      const bids = await req.wallet.getBidsByName(name);
+      const items = [];
+
+      for (const bid of bids) {
+        if (!own || bid.own)
+          items.push(bid.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // All Wallet Reveals
+    this.get('/wallet/:id/reveal', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const own = valid.bool('own', false);
+
+      const reveals = await req.wallet.getRevealsByName();
+      const items = [];
+
+      for (const brv of reveals) {
+        if (!own || brv.own)
+          items.push(brv.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Wallet Reveals by Name
+    this.get('/wallet/:id/reveal/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      let own = valid.bool('own', false);
+
+      if (name)
+        assert(rules.verifyName(name), 'Must pass valid name.');
+
+      if (!name)
+        own = true;
+
+      const reveals = await req.wallet.getRevealsByName(name);
+      const items = [];
+
+      for (const brv of reveals) {
+        if (!own || brv.own)
+          items.push(brv.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Name Resource
+    this.get('/wallet/:id/resource/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
+
+      if (!ns || ns.data.length === 0)
+        return res.json(404);
+
+      try {
+        const resource = Resource.decode(ns.data);
```

Comment:
This means that it is not possible to get the data from the wallet API if it is not encoded properly. This relates to https://github.com/handshake-org/hsd/issues/125

Reply:
@tynes this is only a problem if we allow malformed data to be stored through the standard API.

Reply:
If this is a problem later we can always return `ns.data.toString('hex')` in the `catch` statement